calyx/src/ir/printer.rs

```diff
@@ -160,14 +160,25 @@ impl Printer {
                 matches!(p.borrow().direction, ir::Direction::Output)
             });
 
-        writeln!(
-            f,
-            "component {}{}({}) -> ({}) {{",
-            comp.name.id,
-            Self::format_attributes(&comp.attributes),
-            Self::format_ports(&inputs),
-            Self::format_ports(&outputs),
-        )?;
+        if comp.is_comb {
```

Comment:
Possible to simplify this by having a `let pre = if comp.is_come { "comb " } else { "" };`